pod/perldiag.pod

```diff
@@ -1042,6 +1042,11 @@ error occurs in cases such as these:
     goto G;
     $x + do { G: $y }; # How is + supposed to get its first operand?
 
+=item Can't "goto" into a defer block
```

Comment:
Should `defer` be quoted, for consistency with `"given"` below?

Reply:
I did think about that, but then the other message doesn't have quotes:

```
=item Can't "%s" out of a defer block
```

Basically it's all a random inconsistent mess.

Reply:
Actually, since those messages are new in 5.35.x, I'll do a second commit after this one to just neaten them all up a bit.